ml-agents-envs/setup.py

```diff
@@ -15,16 +15,16 @@
         "Topic :: Scientific/Engineering :: Artificial Intelligence",
         "License :: OSI Approved :: Apache Software License",
         "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
     ],
     packages=["mlagents.envs", "mlagents.envs.communicator_objects"],  # Required
     zip_safe=False,
     install_requires=[
-        "Pillow>=4.2.1,<=5.4.1",
-        "numpy>=1.13.3,<=1.16.1",
-        "pytest>=3.2.2,<4.0.0",
-        "protobuf>=3.6,<3.7",
-        "grpcio>=1.11.0,<1.12.0",
         "cloudpickle",
+        "grpcio>=1.11.0",
+        "numpy>=1.13.3,<2.0",
+        "Pillow>=4.2.1",
+        "protobuf>=3.6",
     ],
-    python_requires=">=3.5,<3.8",
```

Comment:
Why did we remove <3.8 ? Do we support Python 3.8 ? If we only do 3.6 and 3.7, I would prefer keeping an upper bound.

Reply:
3.8 isn't out yet (although RC1 should be out soon according to https://www.python.org/dev/peps/pep-0569/)

For both python versions and dependent library versions (e.g. grpcio), I think we should be as flexible as possible for future versions, unless we know there's a problem. Maybe for major versions (e.g. tensorflow 2.0) we should be conservative since we know they're going to change their interface.

The way I think about it is:
* if someone is using python 3.8 and we allow it, they _might_ have problems. But chances are, it will be compatible and work fine.
* if someone is using python 3.8 and we forbid it, they _definitely_ can't use ml-agents.
I'd prefer the first scenario, and we fix any bugs that arise from it.